types/redux-form/lib/reduxForm.d.ts

```diff
@@ -143,22 +143,26 @@ export interface FormInstance<FormData, P, ErrorType> extends Component<P> {
     submit(): Promise<any>;
     valid: boolean;
     values: Partial<FormData>;
-    wrappedInstance: ReactElement<P & InjectedFormProps<FormData, P, ErrorType>>;
+    wrappedInstance?: HTMLElement;
 }
 
-export interface DecoratedComponentClass<FormData, P, ErrorType> {
-    new(props?: P, context?: any): FormInstance<FormData, P, ErrorType>;
-}
+export type DecoratedFormProps<
+  FormData = {},
+  P = {},
+  ErrorType = string
+> = P & Partial<ConfigProps<FormData, P, ErrorType>> & { [key: string]: any };
```

Comment:
I think we should make whatever other changes are necessary to eliminate this `{ [key: string]: any }`, unless the type truly does have an unknowable string index signature at runtime.